File: cpp/ProjectEulerCpp/ProjectEulerCpp/retos/reto09.cpp
#include "reto09.h"

#include <complex>

int reto09()
{
    for(int a = 3; a<10000; a++)
    {
        for(int b = a; b<10000; b++)
        {
            double c = std::sqrt(a*a + b*b);
            if(floor(c) == c)
            {
                if((a+b+c) == 1000)
                {
                    return a*b*c;
                }
            }
        }
    }
    return 1;
}

File: python/reto09.py
def reto09():
    for a in range(3, 10000):
        for b in range(3, 10000):
            c = (a * a + b * b) ** 0.5
            if c % 1 == 0:
                if a + b + c == 1000:
                    return a * b * c

    return 1
